web/src/components/Course/CourseList.jsx

```diff
@@ -5,7 +5,7 @@ export function CourseList({ courses = [] }) {
     <div>
       {courses.length > 0 ? (
         courses.map((course) => (
-          <CourseDetails key={course.id} course={course} />
+          <CourseDetails key={course.id} course={course.name} />
```

Comment:
You pass a course name here but no id. In the `CourseDetails` component you are trying to get the id from params. That doesn't fit together...🤔